chore/travis/travis-extra.sh

```diff
@@ -33,6 +33,7 @@ git diff
 
 mvn -q test
 mvn -q checkstyle:checkstyle license:check
+mvn -q se.kth.castor:depclean-maven-plugin:2.0.0:depclean -DfailIfUnusedDirect=true
```

Comment:
Isn't `-DfailIfUnusedDirect=true` now duplicated? It's already in the pom.xml?

Reply:
No, it's not. Let me explain.

In `spoon-pom/pom.xml` depclean is added as a Maven profile configured for `spoon-core`, which allows the job to be triggered separately in Travis. 

In `travis-extra.sh`, depclean is executed directly from the command line because this script makes several additional checks in several Spoon extra projects. Using depclean directly gives more flexibility to developers who can configure depclean with [optional parameters](https://github.com/castor-software/depclean#optional-parameters) for each project. For example, to ignore some dependencies that they know are used dynamically or that simply don't want to analyze.

Reply:
Right, I get that, but why not use the depclean profile here as well? I.e.

```suggestion
mvn -q se.kth.castor:depclean-maven-plugin:2.0.0:depclean -Pdepclean
```

I guess I just don't understand why we'd want different configurations.

On a related note, I think what @monperrus meant was to put the depclean job as part of `travis-extra` instead of having it's own dedicated workflow. With this configuration, depclean will run twice, which doesn't seem useful?

Reply:
> With this configuration, depclean will run twice, which doesn't seem useful?

It depends on the decision of the maintainer. I saw a `travis-coverage.sh` so I added depclean job in the Travis matrix.
But of course, having depclean only in `travis-extra` is also an option.

@monperrus Is this what you meant?

Reply:
yes, I meant adding some lines in `travis-extra.sh` instead of adding a new file/job.

BTW: now that we're out of Travis, we could rename `travis*.sh` to `ci-*.sh`

Reply:
@monperrus On that note, we should probably remove all Travis stuff.

Reply:
See #3836